src/main/java/com/slimebot/level/LevelListener.java

```diff
@@ -0,0 +1,96 @@
+package com.slimebot.level;
+
+import com.slimebot.main.Main;
+import com.slimebot.main.config.LevelConfig;
+import com.slimebot.util.MathUtil;
+import com.slimebot.util.Pair;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.concrete.VoiceChannel;
+import net.dv8tion.jda.api.entities.channel.unions.AudioChannelUnion;
+import net.dv8tion.jda.api.events.guild.voice.GenericGuildVoiceEvent;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+public class LevelListener extends ListenerAdapter implements Runnable {
+
+    private final static LevelConfig config = Main.config.level;
+
+    public LevelListener() {
+        try {
+            Main.jdaInstance.awaitReady();
```

Comment:
Remove the awaitReady call, it's not required

Reply:
it is required because i use the discord API right after. to use the discordAPI the bot should be ready

Reply:
Then don't use awaitReady, try to avoid blocking methods. You can instead store the LevelListener instance in main and add a method, that does all the setup code. Then just call this method in the ReadyListener